x/twap/logic.go

```diff
@@ -195,18 +203,6 @@ func (k Keeper) updateRecord(ctx sdk.Context, record types.TwapRecord) (types.Tw
 	return newRecord, nil
 }
 
-// pruneRecords prunes twap records that happened earlier than recordHistoryKeepPeriod
-// before current block time while preserving the most recent record before the threshold.
-// Such record is preserved for each pool.
-// See TWAP keeper's `pruneRecordsBeforeTimeButNewest(...)` for more details about the reasons for
-// keeping this record.
-func (k Keeper) pruneRecords(ctx sdk.Context) error {
-	recordHistoryKeepPeriod := k.RecordHistoryKeepPeriod(ctx)
-
-	lastKeptTime := ctx.BlockTime().Add(-recordHistoryKeepPeriod)
-	return k.pruneRecordsBeforeTimeButNewest(ctx, lastKeptTime)
-}
```

Comment:
This helper func is no longer needed since we determine the lastKeptTime at epoch end and store in a state entry.

Reply:
Seems like we should still keep this function so we can unit test it?

Reply:
Or do you think the current test is good / simple enough?

Reply:
I can reimplement the test, I thought the state.go test tests the same thing, but I can double check

Reply:
Yeah this is tested in even more depth here https://github.com/osmosis-labs/osmosis/blob/81e7ebe79e9e483b8d6512a7f0cb6f53f9c521ab/x/twap/store_test.go#L321-L555

Deleting the func along with the test is fine imo